### DISCUSSION TITLE
Find multiple elements in the page

### DISCUSSION THREAD

**Discussion by montogeek** [keathley/wallaby#670] [Q&A] [answered]
Hi!

How to find multiple elements in the page? I have tried this:
```elixir
session
    |> visit("/")
    |> find(Query.data("test", "title", count: 1, text: "Foo"))
    |> find(Query.data("test", "subTitle", count: 1))
```

But it fails because the first `find` returns the element, so the next `find` won't find anything.
Docs says that you can pass a callback to `find` so it returns the parent element, but I can't find a way to pass an empty callback.

What else can I do?

Thanks!

**Comment by montogeek** [comment]
I found an alternative:
```elixir
|> find(Query.data("test", "title", count: 1, text: "Foo"), fn el -> el end)
```

**Comment by mhanberg** [comment, marked as answer]
It sounds like you may want to assert the existence of an element on the page. To do that you can write

```elixir
session
|> visit("/")
|> assert_has(Query.data("test", "title", count: 1, text: "Foo"))
|> assert_has(Query.data("test", "subTitle", count: 1))
```

You would use `find/2` or `find/3` when you want to retrieve an element to use it to scope further queries or to make assertions on that element.

**Reply by montogeek** [reply]
Thanks!
